src/config.rs

```diff
@@ -0,0 +1,42 @@
+use crate::utils;
+
+use dirs::home_dir;
+
+pub struct Config {
+    data: toml::value::Table,
+}
+
+impl Config {
+    /// Initialize the Config struct
+    pub fn initialize() -> Config {
+        if let Some(file_data) = Config::config_from_file() {
+            return Config { data: file_data };
+        }
+
+        Config {
+            data: toml::value::Table::new(),
+        }
+    }
+
+    /// Create a config from a starship configuration file
+    fn config_from_file() -> Option<toml::value::Table> {
+        let file_path = home_dir()?.join(".config/starship.toml");
```

Comment:
How does one config a config :P Do you think we there is a case for having the config somewhere other than the home dir?

Reply:
This can be handled with an environment variable (`$STARSHIP_CONFIG`).
As was suggested by @Snuggle in https://github.com/starship/starship/issues/10#issuecomment-484329229, we should use `$XDG_CONFIG_HOME`, if available, before resorting to `~/.config`.
This was something I was thinking we would address in a future PR.